### ISSUE TITLE
Get video thumbnail based on hashid without apitoken

### ISSUE THREAD

**Issue by hixfield** [wistia/WistiaKit#57]
Is there a way for getting a video image thumbnail solely based on the hashid of a given video WITHOUT needing an APIToken?

**Comment by spinosa** [comment]
Not with WistiaKit, atm.  I believe the official is yes, and the supported route would be to use the oEmbed endpoint: https://wistia.com/doc/working-with-images#using_oembed

Maybe @jayholman knows of other options..?

**Comment by jayholman** [comment]
That's right @spinosa, the oEmbed endpoint would be the way to do that. That bit of documentation you linked to goes into more detail, but it'll return some JSON that contains the thumbnail's direct asset url.

**Comment by hixfield** [comment]
OK, great I will have a look into that.
It might be a good idea to implement this also in WistiaKit?

**Comment by spinosa** [comment]
@hixfield i'm not opposed to that!  Seems like it should be a class method on `WistiaMedia`

**Comment by hixfield** [comment]
Can't really use a class method because then the sessionManager cannot be reused (its a member var from WistiaAPI). My idea is to then use the init without the APIToken.
Then I tested the following (what do you all think?) :

```
extension WistiaAPI {
    //MARK: - Account
    
    /**
     Get url of the previous image for the movie.
     
     See [Wistia Working with images - https://wistia.com/doc/working-with-images#using_oembed)
     
     - parameter forHash: the hashId that identifies the movie
     - parameter imageWidthPoints: desired width of the image in points (the function converts to pixels taken into account the screen's scale)
     - parameter completionHander: The block to invoke when the API call completes. The block takes 2 arguments: the requested url and the error status code of the request.
     */
    public func getPreviewImageUrl(forHash movieHashedId: String, imageWidthPoints: CGFloat = 640, completionHander: @escaping (_ url: URL?, _ error: WistiaAPIError?) -> () ){
        let widthPixels = Int(imageWidthPoints * UIScreen.main.scale)
        let url = "http://fast.wistia.net/oembed?url=http://home.wistia.com/medias/$$movieHasedId$$?embedType=async&videoWidth=$$imageWidth$$".replace("$$movieHasedId$$", replacement: movieHashedId).replace("$$imageWidth$$", replacement: String(widthPixels))
        sessionManager.request(url, method: .get)
            .validate(statusCode  : 200..<300)
            .responseJSON {
                response in
                switch response.result {
                case .success(let jsonData):
                    guard
                        let JSON = jsonData as? [String: Any],
                        let urlString = JSON["thumbnail_url"] as? String,
                        let url = URL(string: urlString)
                        else {
                            completionHander(nil, WistiaAPIError.InvalidJSON(jsonData))
                            return
                    }
                    log.info("\(String(describing: JSON))")
                    completionHander(url, nil)
                default                    :
                    log.error("\(response)")
                    completionHander(nil, WistiaAPIError.Unknown(nil))
                }
        }
    }
    
}
```

**Comment by spinosa** [comment]
I would like a more consistent API.  Currently one can get a `WistiaMedia.Thumbnail` from a `WistiaMedia` retrieved via an authenticated `WistiaAPI`.  Ideally, one could get a `WistiaMedia.Thumbnail` directly from `WistiaMedia` without an API token.

One approach to accomplish this would be to add a non-authenticated, lazy-loaded `WistiaAPI` singleton (e.g. `WistiaAPI.anonymous()`) and use that within a new class method `WistiaMedia.thumbnailForHashedID(_ hashedID: String, width: Int) -> Thumbnail`

The caller doesn't have to know about `WistiaAPI.anonymous()`, it's an implementation detail.

**Comment by hixfield** [comment]
indeed, I am also for a consistent API and thus implementing WistiaMedia.Thumbnail.
Not sure about the WistiaAPI.anonymous() singleton, seams overhead just to have a sessionManager, it might be better just to implement a local sessionManager no?

**Comment by spinosa** [comment]
two things
1) Maybe the method should actually be an initializer of `WistiaMedia.thumbnail`, seems even cleaner

2) Agreed that the singleton seems a bit heavy for this one thing.  But my gut says there may be other use cases and additional need (for the anonymous API) in the future

**Comment by hixfield** [comment]
1. Including it as part of the initializer it's not a good idea I think. When the initializer completes the object should be immediatly usable.

Therefore I believe the previous idea is better (factory that produces objects pattern). Where its a class method of the WistaAPI.

2. In that case this could it could be a good idea. But it still feels strange to me...

**Comment by spinosa** [comment]
1.  Totally agree.  Factory method with completion handler is the way to go.
2.  thanks for trusting me :-]

**Comment by hixfield** [comment]
implemented, refer to pull request : #58
